OpenRA.Mods.Common/Effects/LineImpactProjectile.cs

```diff
@@ -0,0 +1,256 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.GameRules;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class LineImpactProjectileInfo : IProjectileInfo
+	{
+		[Desc("Projectile speed in WDist / tick,", "two values indicate variable velocity.")]
+		public readonly WDist[] Speed = { new WDist(17) };
+
+		[Desc("The number of ticks between the beam", "causing warhead impacts in its area of effect.")]
+		public readonly int TicksBetweenImpacts = 3;
+
+		[Desc("How far beyond the target the", "projectile keeps on travelling.")]
+		public readonly WDist BeyondTargetRange = new WDist(128);
+
+		[Desc("How close to the target must the", "projectile be to apply weapon effects.")]
+		public readonly WDist MaxTargetRange = new WDist(128);
+
+		[Desc("The maximum reduction to warhead effects", "100 = 100% falloff = 0% damage.")]
+		public readonly int FalloffMaxReduction = 75;
```

Comment:
"Falloff" is a big ambiguous: without reading the code I can't tell whether this is a distance along the line, distance perpendicular to it, or damage falloff for each actor hit.